### ISSUE TITLE
Restore Backup - Bind Issue

### ISSUE THREAD

**Issue by asuweb** [virtualmin/virtualmin-gpl#24]
Restoring a backup from one server to another works correctly, however for some reason the newly restored domain BIND DNS zone is missing from Webmin -> Servers -> BIND DNS.  

The zone file is correctly created, as is the master zone in named.conf, and the records correctly show up in Virtualmin -> Server Configuration -> DNS Records, but they should also be showing up in Webmin, but currently aren't.  

Any ideas?

**Comment by jcameron** [comment]
Yes, they should be showing up there.

Are you logged into Virtualmin/Webmin as `root` ?

**Comment by asuweb** [comment]
Yes, logged in as root.  Strange, they've always shown up there before.

**Comment by asuweb** [comment]
This might be the source of the issue

```
/var/log/webmin/miniserv.error

Use of uninitialized value $_[0] in lc at /usr/libexec/webmin/bind8/index.cgi line 513.
Use of uninitialized value within @sp0 in pattern match (m//) at /usr/libexec/webmin/bind8/index.cgi line 516.
Use of uninitialized value in string comparison (cmp) at /usr/libexec/webmin/bind8/index.cgi line 521.
Use of uninitialized value within @sp0 in pattern match (m//) at /usr/libexec/webmin/bind8/index.cgi line 516.
Use of uninitialized value in string comparison (cmp) at /usr/libexec/webmin/bind8/index.cgi line 521.

```

**Comment by asuweb** [comment]
Strangely, I created a new virtual server to see if it corrected the issue, and it it has.  All zones are now showing up.  

This was a fresh server, prior to the first restore being run.  It seems for some reason that you need to manually create a virtual server, before any restored zones show up in Webmin -> Bind.  

A subsequent restore of another account has worked correctly, and is now showing in bind.

**Comment by jcameron** [comment]
Ok, glad that's working for you now. Those perl warnings seem unrelated, but I'll fix them anyway..

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
